sdk/python/kfp/_client.py

```diff
@@ -277,15 +280,35 @@ def set_user_namespace(self, namespace):
       namespace: kubernetes namespace the user has access to.
     """
     self._context_setting['namespace'] = namespace
+    if not os.path.exists(os.path.dirname(Client.LOCAL_KFP_CONTEXT)):
+        os.makedirs(os.path.dirname(Client.LOCAL_KFP_CONTEXT))
     with open(Client.LOCAL_KFP_CONTEXT, 'w') as f:
       json.dump(self._context_setting, f)
 
+  def get_kfp_healthz(self):
+    """Get healthz info of KFP deployment.
+
+    Returns:
+      response: json formatted response from the healtz endpoint.
+    """
+    healthz_api = 'http://' + Client.IN_CLUSTER_DNS_NAME.format('kubeflow') + '/' + Client.HEALTH_PATH
+    r = requests.get(healthz_api)
+    response = r.json()
+    return response
+
   def get_user_namespace(self):
     """Get user namespace in context config.
 
     Returns:
       namespace: kubernetes namespace from the local context file or empty if it wasn't set.
     """
+    if self._context_setting['namespace'] == "":
+      if self.get_kfp_healthz()['multi_user'] is True:
```

Comment:
Suggest moving the check to Client instantiation, so that it only needs to be called once. Multi user mode should be a constant.

Reply:
+1

Also, maybe do `self.get_kfp_healthz.get("multi_user")` so that it does not break if an old apiserver image is running